blocks/embedded-form/embedded-form.js

```diff
@@ -0,0 +1,152 @@
+import { div } from '../../scripts/scripts.js';
+import { getMetadata } from '../../scripts/lib-franklin.js';
+
+// TODO: submit fetch to the secondary endpoint (currently submitting to the value specified in the action attr of the form element)
+
+export default async function decorate(block) {
+  const cardWrapper = block.querySelector(':scope > div > div:first-child');
+  cardWrapper.classList.add('card-wrapper');
+  await decorateCard(cardWrapper);
+
+  const rightColumnWrapper = block.querySelector(':scope > div > div:last-child');
+  rightColumnWrapper.classList.add('right-column-wrapper');
+
+  await decorateRightColumn(rightColumnWrapper);
+
+  // Fetch and set properties for hidden form fields populated via page metadata
+  document.querySelectorAll('input[type="hidden"]').forEach((field) => {
+    const value = getMetadata(field.name);
+    if (value !== '') {
+      setFormValue(field.id, value);
+    }
+  });
+}
+
+function setFormValue(id, value) {
+  const formElement = document.getElementById(id);
+  if (formElement) {
+    formElement.value = value;
+  }
+}
+
+async function decorateCard(cardWrapper) {
+  const card = div({ class: 'card' });
+  card.append(...cardWrapper.children);
+  cardWrapper.append(card);
+  // remove button containers
+
+  card.querySelectorAll('.button-container').forEach((container) => {
+    [...container.children].forEach((node) => {
+      node.classList.remove('button');
+      node.classList.remove('primary');
+    });
+    container.classList.remove('button-container');
+  });
+}
+
+function validateInput(input) {
+  const formEl = input.closest('.Vlt-form__element');
+  const isFocused = document.activeElement === input;
+  const isValid = input.validity.valid;
+  const isDirty = formEl.classList.contains('Vlt-form__element--dirty');
+
+  if (isFocused) {
+    // remove all alerts when focused
+    formEl.classList.remove('Vlt-form__element--error');
+    formEl.classList.remove('Vlt-form__element--valid');
+    return;
+  }
+
+  if (isValid) {
+    formEl.classList.remove('Vlt-form__element--error');
+  } else {
+    formEl.classList.add('Vlt-form__element--error');
+  }
+
+  if (isValid && isDirty) {
+    formEl.classList.add('Vlt-form__element--valid');
+  } else {
+    formEl.classList.remove('Vlt-form__element--valid');
+  }
+}
+
+function submitForm() {
+  const form = document.querySelector("[data-form-type='lead form: apps: contact sales: in page']");
+  form.querySelectorAll('input[required], select[required]').forEach((input) => {
+    validateInput(input);
+  });
+  // there's a captcha that needs to be integrated. Thus, there will always be one error flag.
+  if (form.querySelectorAll('.Vlt-form__element--error').length <= 1) {
+    // fill in composite form fields
+    setFormValue('phonenumber', document.getElementById('dialing-code').value + document.getElementById('phone-number-local').value);
```

Comment:
can we use `form.getElementById` instead of `document.getElementById`?

Reply:
Yup we sure can and should, will push that change up with the next commit.

Reply:
Scratch that, just found out that apparently I can't :D  Seems like getElementById is only available on the Document level not an individual DOM element.  I'll switch to using a class name and a selector from the forms element

Reply:
ah! you can use 
```
block.querySelector(`#${myvar}`)
```

Reply:
Ah good to know that there is a different way of getting the handle via the ID without climbing all the way up to the document level.  I ended up going with an attribute selector for the time being.  I've got another branch that will be continuing this work so I can switch to that approach instead of there's a good reason to do so :D